File: problem2.py
from io import BytesIO
from PIL import Image

def get_thumbnail(filename):
    with open(filename, 'rb') as f:
        image_content = f.read()
    image_hex = image_content.hex()
    start = 'ffd8ff'
    end = 'ffd9'
    start_index = image_hex.rfind(start)
    end_index = image_hex.find(end)

    thumbnail_hex = image_hex[start_index:end_index+4]
    thumbnail_content = bytes.fromhex(thumbnail_hex)
    thumbnail = Image.open(BytesIO(thumbnail_content))
    thumbnail.show()

def main() :
    filename = 'C:\\Users\\HancomGMD\\Desktop\\1.jpg'
    get_thumbnail(filename)

main()

File: problem2.cs
using System;
using System.Drawing;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.IO;

namespace ConsoleApp1
{

    class problem2
    {
        static public byte[] imageToByteArray(System.Drawing.Image imageIn)
        {
            MemoryStream ms = new MemoryStream();
            imageIn.Save(ms, System.Drawing.Imaging.ImageFormat.Jpeg);
            return ms.ToArray();
        }

        static public Image ByteArrayToImage(byte[] bytes)
        {
            MemoryStream ms = new MemoryStream(bytes);
            Image recImg = Image.FromStream(ms);
            return recImg;
        }

        static public byte[] StringToByteArray(String hex)
        {
            int NumberChars = hex.Length;
            byte[] bytes = new byte[NumberChars / 2];
            for (int i = 0; i < NumberChars; i += 2)
                bytes[i / 2] = Convert.ToByte(hex.Substring(i, 2), 16);
            return bytes;
        }

        static public Image ImageToThumbnail(Image image)
        {
            byte[] image_byte = imageToByteArray(image);
            String image_hex = BitConverter.ToString(image_byte).Replace("-", "");

            String start = "FFD8FF";
            String end = "FFD9";
            int start_index = image_hex.LastIndexOf(start);
            int end_index = image_hex.IndexOf(end);

            String thumbnail_hex = image_hex.Substring(start_index, end_index - start_index + 4);

            byte[] thumbnail_byte = StringToByteArray(thumbnail_hex);
            Image thumbnail = ByteArrayToImage(thumbnail_byte);

            return thumbnail;
        }
        static void Main(string[] args)

        {
            Image image = Image.FromFile("C:\\Users\\HancomGMD\\Desktop\\1.jpg");
            Image thumbnail = ImageToThumbnail(image);
            thumbnail.Save("C:\\Users\\HancomGMD\\Desktop\\thumbnail.jpg");

        }
    }
}
